src/main/java/pedido/Pedido.java

```diff
@@ -27,22 +31,84 @@ public Cliente getCliente(){
     }
 
     public double calcularTotal(Cardapio cardapio){
-        double total= 0;
-        //TODO
-        return total;
+        double precoTotal = 0;
+        for (ItemPedido itemPedido : this.itens){
+            Base basePedido = itemPedido.getShake().getBase();
+            Double precoBase = cardapio.buscarPreco(basePedido);
+
+            switch (itemPedido.getShake().getTipoTamanho()) {
+                case P:
+                    precoTotal += precoBase * itemPedido.getQuantidade();
+                    break;
+                case M:
+                    precoTotal += precoBase * 1.3 * itemPedido.getQuantidade();
+                    break;
+                case G:
+                    precoTotal += precoBase * 1.5 * itemPedido.getQuantidade();
+                    break;
```

Comment:
a lógica do enum poderia ser implementada dentro da classe enum TipoTamanho() evitando o uso do switch case. Exemplo:

    public enum TipoTamanho {
        P (0), M(0.3), G(0.5);
        public final double multiplicador;

        TipoTamanho(final double multiplicador) {
            this.multiplicador = multiplicador;
        }
    }